externs/shaka/queue.js

```diff
@@ -0,0 +1,114 @@
+/*! @license
+ * Shaka Player
+ * Copyright 2016 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+/**
+ * @externs
+ */
+
+
+/**
+ * @typedef {{
+ *   manifestUri: string,
+ *   startTime: ?(number|Date),
+ *   mimeType: ?string,
+ *   config: ?shaka.extern.PlayerConfiguration,
+ *   extraText: ?Array<!shaka.extern.ExtraText>,
+ *   extraThumbnail: ?Array<string>,
+ *   extraChapter: ?Array<!shaka.extern.ExtraChapter>
+ * }}
+ *
+ * @property {string} manifestUri
+ * @property {?(number|Date)} startTime
+ * @property {?string} mimeType
+ * @property {?shaka.extern.PlayerConfiguration} config
+ * @property {?Array<!shaka.extern.ExtraText>} extraText
+ * @property {?Array<string>} extraThumbnail
+ * @property {?Array<!shaka.extern.ExtraChapter>} extraChapter
+ * @exportDoc
+ */
+shaka.extern.QueueItem;
+
+/**
+ * An object that's responsible for all the queue-related logic
+ * in the player.
+ *
+ * @interface
+ * @exportDoc
+ */
+shaka.extern.IQueueManager = class extends EventTarget {
+  /**
+   * @return {!Promise}
+   */
+  destroy() {}
+
+  /**
+   * Called by the Player to provide an updated configuration any time it
+   * changes.
+   *
+   * @param {shaka.extern.QueueConfiguration} config
+   */
+  configure(config) {}
+
+  /**
+   * Returns the current configuration.
+   *
+   * @return {?shaka.extern.QueueConfiguration}
+   */
+  getConfiguration() {}
+
+  /**
+   * Returns the current item.
+   *
+   * @return {?shaka.extern.QueueItem}
+   */
+  getCurrentItem() {}
+
+  /**
+   * Returns the index of the current playing item.
+   *
+   * @return {?number}
+   */
+  getCurrentItemIndex() {}
+
```

Comment:
Just return -1 if no current item, that would be compatible with various JS APIs, like `findIndex()`